pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2370,8 +2401,12 @@ func (n *NodeStatus) IsWorkflowStep() bool {
 }
 
 // Fulfilled returns whether a phase is fulfilled, i.e. it completed execution or was skipped or omitted
-func (phase NodePhase) Fulfilled() bool {
-	return phase.Completed() || phase == NodeSkipped || phase == NodeOmitted
+func (phase NodePhase) Fulfilled(synced *bool) bool {
```

Comment:
Note that Fulfilled is a method on both NodePhase and NodeStatus, they are actually semantically different, this is almost certainly a subtle source of bugs, we should refactor this into one function in a different PR.

Reply:
Could you pop in a Task Issue for this please